core/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.elasticsearch.indices.cache.query.terms.TermsLookup;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+
+public class TermsQueryBuilderTest extends BaseQueryTestCase<TermsQueryBuilder> {
+
+    @Override
+    protected TermsQueryBuilder doCreateTestQueryBuilder() {
+        TermsQueryBuilder query;
+        String fieldName = getRandomFieldName();
+        // terms query or lookup query
+        if (randomBoolean()) {
+            // make between 0 and 5 different values of the same type
+            Object[] values = new Object[randomInt(5)];
+            for (int i = 0; i < values.length; i++) {
+                values[i] = getRandomValueForFieldName(fieldName);
+            }
+            query = new TermsQueryBuilder(fieldName, values);
+        } else {
+            // right now the mock service returns us a list of strings
+            query = new TermsQueryBuilder(STRING_FIELD_NAME);
```

Comment:
when the field is unmapped, there is very little check for terms lookup. The deeper checks happen when the field is mapped string.